tests/mock_crust/client_restrictions.rs

```diff
@@ -151,17 +152,26 @@ fn resend_parts_on_exceeding_limit() {
     }
     poll_and_resend(&mut nodes, &mut clients);
 
-    // `MAX_PARTS / 2` parts will be resent from client.
-    let expect_sent_parts = (2 * MAX_PARTS) as u64;
-    assert_eq!(
-        clients[0].inner.get_user_msg_parts_count(),
-        expect_sent_parts
+    // NOTE: these calculations are somewhat approximate – in reality, the user message
+    // parts seem to be around 20,363 bytes each, not 20,480, so an extra one can
```

Comment:
the user message of immutable_data with max size also have a 48 bytes head.
and divided by MAX_PARTS (52), it means most part will be 20363, with 4 parts to be 20363
once change the calculation to be
```rust
let parts_allowed =
    (SOFT_CAPACITY as f64 /
      ((MAX_IMMUTABLE_DATA_SIZE_IN_BYTES + 48) as f64 /  MAX_PARTS as f64)) as u64;
```
the assert_eq can be put back (abs_diff no longer required)

same to the other test